File: NXNavigationExtensionDemo/UIKit/Feature/Basic/Controllers/ViewController10_ScrollView.m
//
//  ViewController10_ScrollView.m
//  NXNavigationExtensionDemo
//
//  Created by Leo Lee on 2020/10/26.
//

#import <NXNavigationExtension/NXNavigationExtension.h>

#import "ViewController10_ScrollView.h"
#import "RandomColorViewController.h"

#import "UIColor+RandomColor.h"

@interface ViewController10_ScrollView ()


@end

@implementation ViewController10_ScrollView

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self.tableView registerClass:[UITableViewCell class] forCellReuseIdentifier:NSStringFromClass([UITableViewCell class])];
}

- (UIColor *)nx_shadowImageTintColor {
    return [UIColor customColorWithLightModeColor:^UIColor * _Nonnull{
        return [UIColor lightGrayColor];
    } darkModeColor:^UIColor * _Nonnull{
        return [[UIColor lightGrayColor] colorWithAlphaComponent:0.65];
    }];
}

#pragma mark - Table view data source

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return 30;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:NSStringFromClass([UITableViewCell class])];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:NSStringFromClass([UITableViewCell class])];
    }
    cell.textLabel.text = [NSString stringWithFormat:@"Row: %02zd", indexPath.row + 1];
    cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    
    cell.textLabel.textColor = [UIColor customColorWithLightModeColor:^UIColor * _Nonnull{
        return [UIColor blackColor];
    } darkModeColor:^UIColor * _Nonnull{
        return [UIColor whiteColor];
    }];
    
    cell.backgroundColor = [UIColor customColorWithLightModeColor:^UIColor * _Nonnull{
        return [UIColor randomLightColor];
    } darkModeColor:^UIColor * _Nonnull{
        return [UIColor randomDarkColor];
    }];;
    
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    [self.navigationController pushViewController:[[RandomColorViewController alloc] init] animated:YES];
}

@end

File: Example/NXNavigationExtensionSwiftDemo/Basic/ViewController10_ScrollView.swift
//
//  ViewController10_ScrollView.swift
//  NXNavigationExtensionSwiftDemo
//
//  Created by Leo Lee on 2020/11/23.
//

import UIKit

class ViewController10_ScrollView: UITableViewController {
    
    private static let reuseIdentifier = String(describing: UITableViewCell.self)

    override func viewDidLoad() {
        super.viewDidLoad()

        tableView.register(UITableViewCell.self, forCellReuseIdentifier: ViewController10_ScrollView.reuseIdentifier)
    }
    
    override var nx_shadowImageTintColor: UIColor? {
        return UIColor.customDynamic(lightMode: { UIColor.lightGray }, darkMode: { UIColor.lightGray.withAlphaComponent(0.65) })
    }
    
}

// MARK: - Table view data source

extension ViewController10_ScrollView {

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 30
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let reuseIdentifier = ViewController10_ScrollView.reuseIdentifier
        let cell = tableView.dequeueReusableCell(withIdentifier: reuseIdentifier) ?? UITableViewCell(style: .default, reuseIdentifier: reuseIdentifier)
        cell.textLabel?.text = String(format: "Row: %02zd with UIScrollView", indexPath.row + 1)
        cell.textLabel?.textColor = .customText
        cell.backgroundColor = UIColor.customDynamic(lightMode: { .randomLight }, darkMode: { .randomDark })
        cell.accessoryType = .disclosureIndicator
        
        return cell
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        navigationController?.pushViewController(RandomColorViewController(), animated: true)
    }

}
